inbox/api/wsgi.py

```diff
@@ -16,130 +14,81 @@
-            status = abs(status)
-
-        log.info(
-            "request handled",
-            response_bytes=length,
-            request_time=request_time,
-            remote_addr=client_address,
-            http_status=status,
-            http_request=requestline,
-            request_method=method,
-            **additional_context
-        )
-
-    def get_environ(self):
-        env = super().get_environ()
-        env["gunicorn.sock"] = self.socket
-        env["RAW_URI"] = self.path
-        return env
-
-    def handle_error(self, type, value, tb):
-        # Suppress tracebacks when e.g. a client disconnects from the streaming
-        # API.
-        if (
-            issubclass(type, socket.error)
-            and value.args[0] == errno.EPIPE
-            and self.response_length
-        ):
-            self.server.log.info("Socket error", exc=value)
-            self.close_connection = True
-        else:
-            super().handle_error(type, value, tb)
-
-
 class NylasWSGIWorker(GeventWorker):
     """Custom worker class for gunicorn. Based on
     gunicorn.workers.ggevent.GeventPyWSGIWorker.
     """
 
-    server_class = WSGIServer
-    wsgi_handler = NylasWSGIHandler
-
     def init_process(self):
         print("Python", sys.version, file=sys.stderr)
 
+        maybe_enable_rollbar()
+
+        configure_logging(log_level=LOGLEVEL)
+
         if MAX_BLOCKING_TIME:
             self.tracer = Tracer(max_blocking_time=MAX_BLOCKING_TIME)
             self.tracer.start()
         super().init_process()
 
 
-class NylasGunicornLogger(gunicorn.glogging.Logger):
-    def __init__(self, cfg):
-        gunicorn.glogging.Logger.__init__(self, cfg)
-        configure_logging(log_level=LOGLEVEL)
-        self.error_log = log
-
-
 from inbox.config import config
 from inbox.error_handling import maybe_enable_rollbar
 
 MAX_BLOCKING_TIME = config.get("MAX_BLOCKING_TIME", 1.0)
 LOGLEVEL = config.get("LOGLEVEL", 10)
 
-# legacy names for backcompat
-InboxWSGIWorker = NylasWSGIWorker
-GunicornLogger = NylasGunicornLogger
 
+class JsonRequestFormatter(json_log_formatter.JSONFormatter):
+    """Custom JSON log formatter for gunicorn access logs.
 
-class RollbarWSGIWorker(NylasWSGIWorker):
-    def init_process(self):
-        maybe_enable_rollbar()
+    Adapted from https://til.codeinthehole.com/posts/how-to-get-gunicorn-to-log-as-json/
+    """
 
-        super().init_process()
+    def json_record(
+        self,
+        message: str,
+        extra: "dict[str, str | int | float]",
+        record: logging.LogRecord,
+    ) -> "dict[str, str | int | float]":
+        # Convert the log record to a JSON object.
+        # See https://docs.gunicorn.org/en/stable/settings.html#access-log-format
+
+        url = record.args["U"]
+        if record.args["q"]:
+            url += f"?{record.args['q']}"
+
+        method = record.args["m"]
+        log_context = record.args.get("{log_context}e", {})
```

Comment:
Yes `{log_context}e`, where `e` stands for WSGI environment. These are "extras" we attach on the fly in the Flask views.